src/libsyntax/feature_gate.rs

```diff
@@ -867,6 +867,8 @@ impl<'a, 'v> Visitor<'v> for PostExpansionVisitor<'a> {
                 self.context.span_handler.span_err(span, "empty tuple structs and enum variants \
                                                           are not allowed, use unit structs and \
                                                           enum variants instead");
+                self.context.span_handler.span_help(span, "remove trailing `()` if you meant to \
+                                                           write empty enum variant or structure");
```

Comment:
The language here is a little bit confusing I would change to "remove trailing `()` to make a unit struct or unit enum variant" as the terminology then matches the error message.